tests/irdl/test_attribute_definition.py

```diff
@@ -684,13 +683,11 @@ def verify(
         self,
         attr: Attribute,
         constraint_context: ConstraintContext,
-    ) -> None:
-        if not isa(attr, ListData):
-            raise VerifyException(
-                f"Expected {attr} to be instance of {ListData.__name__}"
-            )
+    ) -> ListData[AttributeInvT]:
+        attr = BaseAttr(ListData).verify(attr, constraint_context)
```

Comment:
one potential use-case for this pattern is to do multiple stages of verification less verbosely